GoogleLoginPage: fix approval page handling

v5.106
```
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException

	at teammates.test.pageobjects.AppPage.getNewPageInstance(AppPage.java:135)
	at teammates.test.pageobjects.AppPage.createCorrectLoginPageType(AppPage.java:167)
	at teammates.test.cases.browsertests.StudentCourseJoinConfirmationPageUiTest.testJoinNewConfirmation(StudentCourseJoinConfirmationPageUiTest.java:74)
	at teammates.test.cases.browsertests.StudentCourseJoinConfirmationPageUiTest.testAll(StudentCourseJoinConfirmationPageUiTest.java:51)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:659)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:845)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1153)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:771)
	at org.testng.TestRunner.run(TestRunner.java:621)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1199)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1124)
	at org.testng.TestNG.run(TestNG.java:1032)
	at org.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:72)
	at org.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:127)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at teammates.test.pageobjects.AppPage.getNewPageInstance(AppPage.java:131)
	... 31 more
Caused by: java.lang.IllegalStateException: Not in the correct page!
	at teammates.test.pageobjects.AppPage.<init>(AppPage.java:112)
	at teammates.test.pageobjects.LoginPage.<init>(LoginPage.java:6)
	at teammates.test.pageobjects.GoogleLoginPage.<init>(GoogleLoginPage.java:26)
	... 36 more
```

--- a/src/test/java/teammates/test/pageobjects/GoogleLoginPage.java
+++ b/src/test/java/teammates/test/pageobjects/GoogleLoginPage.java
@@ -81,10 +81,25 @@ public class GoogleLoginPage extends LoginPage {
 
     private void handleApprovalPageIfAny() {
         waitForPageToLoad();
+        waitForRedirectIfAny();
         boolean isPageRequestingAccessApproval = getPageSource().contains(EXPECTED_SNIPPET_APPROVAL);
         if (isPageRequestingAccessApproval) {
             click(By.id("persist_checkbox"));
             click(By.id("approve_button"));
+            waitForPageToLoad();
+        }
+    }
+
+    private void waitForRedirectIfAny() {
+        By metaRefreshBy = By.cssSelector("meta[http-equiv='refresh']");
+        if (isElementPresent(metaRefreshBy)) {
+            waitForElementToDisappear(metaRefreshBy);
+            waitForPageToLoad();
+        }
+
+        By noScriptBy = By.tagName("noscript");
+        if (isElementPresent(noScriptBy)) {
+            waitForTextContainedInElementAbsence(noScriptBy, "&lt;meta http-equiv=\"refresh\"");
             waitForPageToLoad();
         }
     }

